test/functional/test_framework/test_framework.py

```diff
@@ -702,57 +704,33 @@ def wait_for_instantlock(self, txid, node):
         return locked
 
     def wait_for_chainlocked_block(self, node, block_hash, timeout=15):
-        t = time.time()
-        while time.time() - t < timeout:
+        def check_chainlocked_block():
             try:
                 block = node.getblock(block_hash)
-                if block["confirmations"] > 0 and block["chainlock"]:
-                    return
+                return block["confirmations"] > 0 and block["chainlock"]
             except:
-                # block might not be on the node yet
-                pass
-            time.sleep(0.1)
-        raise AssertionError("wait_for_chainlocked_block timed out")
+                return False
+        wait_until(check_chainlocked_block, timeout=timeout, sleep=0.1)
 
     def wait_for_chainlocked_tip(self, node):
         tip = node.getbestblockhash()
         self.wait_for_chainlocked_block(node, tip)
 
     def wait_for_chainlocked_tip_all_nodes(self):
         for node in self.nodes:
-            tip = node.getbestblockhash()
-            self.wait_for_chainlocked_block(node, tip)
+            self.wait_for_chainlocked_tip(node)
```

Comment:
This is a change in behavior. Old code ensured that all were on the same tip and had it chainlocked while the new code only verifies that the nodes are at *some* chainlocked tip

Reply:
Hmm... not really imo. This would be the case if we had `sync_blocks` and `tip = node.getbestblockhash()` outside of the loop for example but that's not the case.

Reply:
Ahh sorry, somehow mixed up where tip= was. But...then the whole method was bogus before already as it might return without error even if all nodes are on a different tip, e.g. when called [here](https://github.com/dashpay/dash/blob/develop/test/functional/llmq-chainlocks.py#L42).

Maybe we should remove that whole function as it's not immediately clear what the expected behavior should be? And instead introduce `wait_for_chainlocked_block_all_nodes` where we explicitly say which block to look for a  chainlock?

Reply:
Well, THAT would be a change in behaviour and I'd like to keep this PR refactoring only ;) I do agree that we need to fix it (in another PR) though.

Reply:
Hehe, agree...I'll make it part of #3129 then (with changing the title)